File: android-lib/src/androidTest/java/com/kinvey/androidTest/util/RealmCacheManagerUtil.java
package com.kinvey.androidTest.util;

import com.kinvey.android.Client;
import com.kinvey.android.cache.RealmCacheManager;

import java.lang.reflect.InvocationTargetException;
import java.lang.reflect.Method;

import io.realm.DynamicRealm;

/**
 * Created by yuliya on 11/17/17.
 */

public class RealmCacheManagerUtil {

    public static DynamicRealm getRealm(Client client) {
        Method method = null;
        try {
            method = RealmCacheManager.class.getDeclaredMethod("getDynamicRealm", null);
        } catch (NoSuchMethodException e) {
            e.printStackTrace();
        }
        method.setAccessible(true);
        DynamicRealm realm = null;
        try {
            realm = (DynamicRealm) method.invoke(new RealmCacheManager(client) , null);
        } catch (IllegalAccessException e) {
            e.printStackTrace();
        } catch (InvocationTargetException e) {
            e.printStackTrace();
        }
        return realm;
    }

    public static void setTableHash(Client client, String collection, String hash, DynamicRealm realm) {
        Method method = null;
        try {
            method = RealmCacheManager.class.getDeclaredMethod("setTableHash", String.class, String.class, DynamicRealm.class);
        } catch (NoSuchMethodException e) {
            e.printStackTrace();
        }
        method.setAccessible(true);
        try {
            method.invoke(new RealmCacheManager(client) , collection, hash, realm);
        } catch (IllegalAccessException e) {
            e.printStackTrace();
        } catch (InvocationTargetException e) {
            e.printStackTrace();
        }
    }

}

File: android-lib/src/androidTest/java/com/kinvey/androidTest/util/RealmCacheManagerUtil.kt
package com.kinvey.androidTest.util

import com.kinvey.android.Client
import com.kinvey.android.cache.RealmCacheManager
import io.realm.DynamicRealm
import java.lang.reflect.InvocationTargetException
import java.lang.reflect.Method

/**
 * Created by yuliya on 11/17/17.
 */

object RealmCacheManagerUtil {

    fun getRealm(client: Client<*>?): DynamicRealm? {
        var method: Method? = null
        try {
            method = RealmCacheManager::class.java.getDeclaredMethod("getDynamicRealm", null)
        } catch (e: NoSuchMethodException) {
            e.printStackTrace()
        }
        method?.isAccessible = true
        var realm: DynamicRealm? = null
        try {
            realm = method?.invoke(RealmCacheManager(client!!), null) as DynamicRealm
        } catch (e: IllegalAccessException) {
            e.printStackTrace()
        } catch (e: InvocationTargetException) {
            e.printStackTrace()
        }
        return realm
    }

    fun setTableHash(client: Client<*>?, collection: String?, hash: String?, realm: DynamicRealm?) {
        var method: Method? = null
        try {
            method = RealmCacheManager::class.java.getDeclaredMethod("setTableHash", String::class.java, String::class.java, DynamicRealm::class.java)
        } catch (e: NoSuchMethodException) {
            e.printStackTrace()
        }
        method?.isAccessible = true
        try {
            method?.invoke(RealmCacheManager(client!!), collection, hash, realm)
        } catch (e: IllegalAccessException) {
            e.printStackTrace()
        } catch (e: InvocationTargetException) {
            e.printStackTrace()
        }
    }
}